File: Demo/Sources/ViewControllers/ImagePickerCrop/ImagePickerCropViewController.m
//
//  Globus
//

#import "ImagePickerCropViewController.h"

@interface ImagePickerCropViewController ()

@property(nonatomic, weak) IBOutlet UIImageView* pickedImageView;
@property(nonatomic, weak) IBOutlet UIImageView* croppedImageView;

@end

@implementation ImagePickerCropViewController

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)pressedChoice:(id)sender {
    __weak typeof(self) weakSelf = self;
    GLBImagePickerController* imagePicker = [GLBImagePickerController imagePickerControllerWithViewController:self];
    [imagePicker presentAnimated:YES completion:^(UIImage* image) {
        if(image != nil) {
            weakSelf.pickedImageView.image = image;
            weakSelf.croppedImageView.image = nil;
            
            [GLBTimeout executeBlock:^{
                GLBImageCropViewController* cropViewController = [[GLBImageCropViewController alloc] initWithImage:image];
                cropViewController.choiceBlock = ^(GLBImageCropViewController* cropViewController, UIImage* croppedImage) {
                    weakSelf.croppedImageView.image = croppedImage;
                    [cropViewController dismissViewControllerAnimated:YES completion:nil];
                };
                cropViewController.cancelBlock = ^(GLBImageCropViewController* cropViewController) {
                    [cropViewController dismissViewControllerAnimated:YES completion:nil];
                };
                [self presentViewController:cropViewController animated:YES completion:nil];
            } afterDelay:0.5f];
        } else {
            weakSelf.pickedImageView.image = nil;
            weakSelf.croppedImageView.image = nil;
        }
    }];
}

@end

File: Demo/Sources/ViewControllers/ImagePickerCrop/ImagePickerCropViewController.swift
//
//  Globus
//

import Globus

class ImagePickerCropViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var pickedImageView: UIImageView!
    @IBOutlet fileprivate weak var croppedImageView: UIImageView!
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func pressedChoice(_ sender: Any) {
        let imagePicker = GLBImagePickerController.init(viewController: self)
        imagePicker.present(animated: true) { [unowned self] (image: UIImage?) in
            if (image != nil) {
                self.pickedImageView.image = image
                self.croppedImageView.image = nil
                
                GLBTimeout.execute({ [unowned self] in ()
                    let cropViewController = GLBImageCropViewController.init(image: image!, cropMode: .circle)
                    cropViewController.choiceBlock = { [unowned self] (cropViewController: GLBImageCropViewController, croppedImage: UIImage?) in ()
                        self.croppedImageView.image = croppedImage
                        cropViewController.dismiss(animated: true, completion: nil)
                    }
                    cropViewController.cancelBlock = { (cropViewController: GLBImageCropViewController) in ()
                        cropViewController.dismiss(animated: true, completion: nil)
                    }
                    self.present(cropViewController, animated: true, completion: nil)
                }, afterDelay: 0.5)
            } else {
                self.pickedImageView.image = nil
                self.croppedImageView.image = nil
            }
        }
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "ImagePickerCropViewController-Swift"
    }
    
}
